vllm/v1/worker/gpu_model_runner.py

```diff
@@ -856,34 +875,21 @@ def execute_model(
         # TODO(woosuk): The following loop can be slow since it iterates over
         # the requests one by one. Optimize.
         num_reqs = self.input_batch.num_reqs
-        request_seq_lens: List[Tuple[int, CachedRequestState, int]] = []
-        for i, req_id in enumerate(  # type: ignore[assignment]
-                self.input_batch.req_ids[:num_reqs]):
+        req_ids: List[str] = []
+        for i, req_id in zip(range(num_reqs), self.input_batch.req_ids):
```

Comment:
Why not just use enumerate? Isn't the num_reqs not always equal to len(req_ids)?

```suggestion
        for i, req_id in enumerate(self.input_batch.req_ids):
```

Reply:
Unfortunately, `req_ids` is a list of size `max_num_reqs` at the moment. I think @njhill has a PR to fix this, but I didn't include it in this PR.

Reply:
Might be better to comment it, as using zip to truncate a list for a loop is a bit tricky

Reply:
sg. Added in [c833429](https://github.com/vllm-project/vllm/pull/13339/commits/c833429ceae6748a05a9a4c06bd226e074f22843)